src/middlewared/middlewared/plugins/nfs.py

```diff
@@ -61,3 +64,233 @@ def nfs_compress(self, nfs):
+        paths = data.pop("paths")
+        data["id"] = await self.middleware.call(
+            "datastore.insert", self._config.datastore, data,
+            {
+                "prefix": self._config.datastore_prefix
+            },
+        )
+        for path in paths:
+            await self.middleware.call(
+                "datastore.insert", "sharing.nfs_share_path",
+                {
+                    "share_id": data["id"],
+                    "path": path,
+                },
+            )
+        await self.extend(data)
+
+        await self.middleware.call("service.reload", "nfs")
+
+        return data
+
+    @accepts(
+        Int("id"),
+        Patch(
+            "sharingnfs_create",
+            "sharingnfs_update",
+            ("attr", {"update": True})
+        )
+    )
+    async def do_update(self, id, data):
+        verrors = ValidationErrors()
+        old = await self.middleware.call(
+            "datastore.query", self._config.datastore, [("id", "=", id)],
+            {
+                "extend": self._config.datastore_extend,
+                "prefix": self._config.datastore_prefix,
+                "get": True
+            },
+        )
+
+        new = old.copy()
+        new.update(data)
+
+        await self.validate(new, "sharingnfs_update", verrors, old=old)
+
+        if verrors:
+            raise verrors
+
+        await self.compress(new)
+        paths = new.pop("paths")
+        await self.middleware.call(
+            "datastore.update", self._config.datastore, id, new,
+            {
+                "prefix": self._config.datastore_prefix
+            }
+        )
+        await self.middleware.call("datastore.sql", "DELETE FROM sharing_nfs_share_path WHERE share_id = ?", (id,))
+        for path in paths:
+            await self.middleware.call(
+                "datastore.insert", "sharing.nfs_share_path",
+                {
+                    "share_id": id,
+                    "path": path,
+                },
+            )
+
+        await self.extend(new)
+        new["paths"] = paths
+
+        await self.middleware.call("service.reload", "nfs")
+
+        return new
+
+    @accepts(Int("id"))
+    async def do_delete(self, id):
+        await self.middleware.call("datastore.sql", "DELETE FROM sharing_nfs_share_path WHERE share_id = ?", (id,))
+        await self.middleware.call("datastore.delete", self._config.datastore, id)
+
+    @private
+    async def validate(self, data, schema_name, verrors, old=None):
+        if not data["paths"]:
+            verrors.add(f"{schema_name}.paths", "At least one path is required")
+
+        await self.middleware.run_in_io_thread(self.validate_paths, data, schema_name, verrors)
+
+        if not data["networks"]:
+            verrors.add(f"{schema_name}.networks", "At least one network is required")
+
+        for i, network1 in enumerate(data["networks"]):
+            network1 = ipaddress.ip_network(network1, strict=True)
```

Comment:
We were not being strict with network before, shouldnt we do strict=False here as well?